### PR TITLE
fix(remix-architect): set request URL scheme to `http` when running arc sandbox

### LINKED ISSUE #1536: [Bug]: remix-architect hard codes URL scheme in requests to https, even for dev / sandbox server where requests are made using http
### Which Remix packages are impacted?

- `@remix-run/architect`

### What version of Remix are you using?

1.1.1

### What version of Node are you using? Minimum supported version is 14.

14.18.2

### Steps to Reproduce

1. Start an architect project using create-remix.
2. Start `arc sandbox`.
3. Start `npm run dev`.
4. Retrieve `request.url` from any loader.

### Expected Behavior

The URL scheme should be `http://`.

### Actual Behavior

The URL will start with `http://`, even though the actual request was made using `http://`, because the URL scheme is hardcoded in: https://github.com/remix-run/remix/blob/776cb79b97bdfba42f802c2f583b1ec5d4dcb785/packages/remix-architect/server.ts#L116

### PR DESCRIPTION
**Pull request by lpsinger**
Fixes #1536.

### CODE REVIEW

File: packages/remix-architect/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "@remix-run/server-runtime": "1.1.0"
   },
   "peerDependencies": {
-    "@architect/architect": "^8.3.7"
+    "@architect/architect": "^9.3.1"
```
Comment (jacob-ebey) [inline packages/remix-architect/package.json:20]: Please back out the version update on this PR and we'll get the functionality merged.

**Review by jacob-ebey — CHANGES_REQUESTED** [review]

File: packages/remix-architect/package.json (hunk 1 above)
Comment (lpsinger) [reply packages/remix-architect/package.json:20]: As the commit message says (63590355ec19c6aa42381077fc5750cf918a2cf5):

> Bump architect version to 9.3.1
>
> This version adds a dependency on @architect/sandbox 4.3.1, which added the `ARC_SANDBOX` environment variable.

I found that `"@architect/architect": "^9.3.1"` is the earliest version that pull in @architect/sandbox 4.3.1.

**Comment by lpsinger** [comment]
I switched the base branch to dev.

**Comment by ryanflorence** [comment]
@jacob-ebey Looks like we need the version bump to get `ARC_SANDBOX`, yes?

@lpsinger can you get up to dev and fix the merge conflicts, sorry about that :|

**Comment by lpsinger** [comment]
> @lpsinger can you get up to date with dev and fix the merge conflicts, sorry about that :|

Done. I don't see any merge conflicts.

**Review by mcansh — APPROVED** [review]

**Comment by MichaelDeBoey** [comment]
@lpsinger Could you please rebase onto latest `dev` & sign the CLI?

**Comment by lpsinger** [comment]
> @lpsinger Could you please rebase onto latest `dev` & sign the CLI?

Rebased. I _think_ that since the peerDependency on architect was removed, there are no changes needed to package.json.

As for the CLA, I had already signed it.

**Comment by mcansh** [comment]
> As for the CLA, I had already signed it.

you are correct, label is there and i manually re-verified the contributors list

**Pull request merged**